common/src/main/java/com/wynntils/services/mapdata/MapFeaturePoiWrapper.java

```diff
@@ -63,17 +64,22 @@ public String getName() {
 
     @Override
     public int getWidth(float mapZoom, float scale) {
-        String iconId = attributes.getIconId();
-        String label = attributes.getLabel();
+        Optional<String> iconId = attributes.getIconId();
+        Optional<String> label = attributes.getLabel();
 
-        if (hasIcon(iconId)) {
-            MapIcon icon = Services.MapData.getIcon(iconId).get();
-            return (int) (icon.getWidth() * scale);
+        if (hasIcon()) {
+            Optional<MapIcon> icon = Services.MapData.getIcon(iconId.get());
+            if (icon.isPresent()) {
+                return (int) (icon.get().getWidth() * scale);
+            }
         }
 
-        if (hasLabel(label)) {
+        if (hasLabel(label.orElse(null))) {
             // Use label for measurements
-            return (int) (FontRenderer.getInstance().getFont().width(attributes.getLabel()) * scale);
+            return (int) (FontRenderer.getInstance()
+                            .getFont()
+                            .width(attributes.getLabel().orElse(""))
```

Comment:
I wonder if `attributes.getLabel().orElse("")` is something that should be as a helper function. Not sure where, though. But it sort of feels like it would make sense to have a way to get a guarenteed string, even if it is empty. Otoh, if this is the only place this operation is done, then there is perhaps no big point.

Reply:
If the need comes up when I'm implementing anything related, I'll add the method.